### PR TITLE
Release/v0.1.22

### PR DESCRIPTION
**Pull request by siller174**
1. New functionality `Retry` test
2. Rename functions `RequestRepeat*` to `RequestRetry*`
3. Fix broken tests

### CODE REVIEW

**Comment by siller174** [comment]
@samybenatt hello. I hope you are doing well. Could you please review this PR?

**Review by samybenatt — COMMENTED** [review]
Hey @siller174 
I reviewed all your implementation

I think one key difference between your implementation and mine is that, when one of the attempt is successful, I mark the whole test as successful, and I don't keep error history. I am only failing if the last attempt is still failing.

In your implementation, the test will start failing even after retrying.

The optional and broken error options are not really fulfilling the intended purpose because it is  either setting all failures as broken/optional or all failures as failed. But the retry purpose is the following:
- The test will be retried up to `MaxAttempts` times
- The retries will only be executed if the test is having errors
- If the test is successful at any iteration between attempt 1 and `MaxAttempts`, the loop will break and return **the overall test result as successful** (it will not process previous errors and it will never trigger t.Fail())
- The status of the overall test (success or fail) will be based on either **the first attempt that is successful**, or, if no attempt is successful, **it will be based on the latest execution result**

File: examples/table_test/table_test.go [hunk 1]
```diff
@@ -152,3 +193,205 @@ func Test_Array(t *testing.T) {
...
+					cute.WithMethod(http.MethodGet),
+				},
+			},
+			Expect: &cute.Expect{
+				Code:        202,
+				ExecuteTime: 2,
+			},
+		},
+	}
+
```
[the remark this answers is not in the document]
Comment (siller174) [reply examples/table_test/table_test.go:393]: done

File: test.go [hunk 2]
```diff
@@ -162,52 +190,103 @@ func (it *Test) initEmptyFields() {
...
+		resp, errs = it.startTest(ctx, t)
+
+		resultState = it.processTestErrors(t, errs)
+
+		// if the test is successful, we break the loop
+		if resultState == ResultStateSuccess {
+			break
+		}
+
+		// we want to keep the errors from the previous attempt, so we append them to the new errors
```
[the remark this answers is not in the document]
Comment (siller174) [reply test.go]: got it

**Pull request merged**